### DISCUSSION TITLE
mouse and keyboard not working

### DISCUSSION THREAD

**Discussion by wforumw** [ltsp/ltsp#279] [Q&A]
Hi,
Since the last update to ltsp version 20.04-1~ubuntu18.04.1 the mouse and keyboard is not working in a newly create image.
The mouse and keyboard is still working in an image created with a previous version of ltsp
Any idea why this could happen?
Thanks!!

**Comment by alkisg** [comment]
I transferred this issue to ltsp/community, as it's not a bug in the LTSP code.

I have seen two reasons causing this:

1) To see the client kernel version, run `uname -r` via epoptes. So, if the client booted with e.g. kernel 5.3, but the image doesn't contain the kernel 5.3 modules (`du -sh /lib/modules/*`), and only has e.g. the kernel 5.4 modules, then it'll have this issue.

2) On a 18.04 to 20.04 upgrade, the xserver-xorg-input-all package got uninstalled, so Xorg couldn't detect any input devices.

**Comment by wforumw** [comment]
thanks for the quick reply
The kernel running in the image is 4.19.0-9-amd64
and the package xserver-xorg-input-all (7.7+19) is installed
We are using Debian Buster

**Comment by alkisg** [comment]
What is the output of these commands, on the client? Don't run them on the server, run them via epoptes on the client:

```shell
uname -r
dpkg -l | grep linux
dpkg -l | grep xserver-xorg-input-all
cat /etc/X11/Xorg.0.log
```

**Comment by wforumw** [comment]
Hi,
All packages have been installed.
Very strange, it works now.
Thanks
